kratos/utilities/math_utils.h

```diff
@@ -777,284 +1080,439 @@ class MathUtils
      * @param rStrainVector the given strain vector
      * @return the corresponding strain tensor in matrix form
      */
+    
     static inline MatrixType StrainVectorToTensor( const VectorType& rStrainVector)
     {
-      KRATOS_TRY
-      Matrix StrainTensor;
+        KRATOS_TRY
+        Matrix StrainTensor;
 
-      if (rStrainVector.size()==3)
+        if (rStrainVector.size()==3)
         {
-	  StrainTensor.resize(2,2, false);
-	  StrainTensor(0,0) = rStrainVector[0];
-	  StrainTensor(0,1) = 0.5*rStrainVector[2];
-	  StrainTensor(1,0) = 0.5*rStrainVector[2];
-	  StrainTensor(1,1) = rStrainVector[1];
+            StrainTensor.resize(2,2, false);
+            StrainTensor(0,0) = rStrainVector[0];
+            StrainTensor(0,1) = 0.5*rStrainVector[2];
+            StrainTensor(1,0) = 0.5*rStrainVector[2];
+            StrainTensor(1,1) = rStrainVector[1];
         }
-      else if (rStrainVector.size()==4)
+        else if (rStrainVector.size()==4)
         {
-	  StrainTensor.resize(3,3, false);
-	  StrainTensor(0,0) = rStrainVector[0];
-	  StrainTensor(0,1) = 0.5*rStrainVector[3];
-	  StrainTensor(0,2) = 0;
-	  StrainTensor(1,0) = 0.5*rStrainVector[3];
-	  StrainTensor(1,1) = rStrainVector[1];
-	  StrainTensor(1,2) = 0;
-	  StrainTensor(2,0) = 0;
-	  StrainTensor(2,1) = 0;
-	  StrainTensor(2,2) = rStrainVector[2];
+            StrainTensor.resize(3,3, false);
+            StrainTensor(0,0) = rStrainVector[0];
+            StrainTensor(0,1) = 0.5*rStrainVector[3];
+            StrainTensor(0,2) = 0;
+            StrainTensor(1,0) = 0.5*rStrainVector[3];
+            StrainTensor(1,1) = rStrainVector[1];
+            StrainTensor(1,2) = 0;
+            StrainTensor(2,0) = 0;
+            StrainTensor(2,1) = 0;
+            StrainTensor(2,2) = rStrainVector[2];
         }
-      else if (rStrainVector.size()==6)
+        else if (rStrainVector.size()==6)
         {
-	  StrainTensor.resize(3,3, false);
```

Comment:
here it should be:

if(StrainTenstor.size1() != 3, StrainTensor.size2() != 3)
             StrainTensor.resize(3,3,false);

(clearly this is only an improvement....)